pkg/identity/certificated.go

```diff
@@ -45,21 +45,29 @@ func Certificated(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (er
 		log.Infof("Role certificate provisioning is disabled with empty options: roles[%s], output directory[%s]", idConfig.TargetDomainRoles, idConfig.RoleCertDir)
 	}
 
-	var identity, k8sSecretBackupIdentity, forceInitIdentity *InstanceIdentity
-	var keyPEM, k8sSecretBackupKeyPEM, forceInitKeyPEM []byte
-
 	handler, err := InitIdentityHandler(idConfig)
 	if err != nil {
 		log.Errorf("Failed to initialize client for certificates: %s", err.Error())
 		return err, nil
 	}
 
-	writeFiles := func(id *InstanceIdentity, keyPEM []byte, roleCerts [](*RoleCertificate), roleKeyPEM []byte) error {
+	// identity & keyPEM will be STORED to the local file system:
+	var keyPEM, k8sSecretBackupKeyPEM, forceInitKeyPEM []byte
+	var identity, k8sSecretBackupIdentity, forceInitIdentity *InstanceIdentity
+
+	// RoleCert Keys and Certs will be STORED to the local file system:
+	var roleKeyPEM []byte
+	var roleCerts [](*RoleCertificate)
 
-		w := util.NewWriter()
+	// identity & keyPEM that will NOT be STORED to the local file system:
+	var localFileKeyPEM []byte
+	var localFileIdentity  *InstanceIdentity
 
-		if id != nil {
-			leafPEM := []byte(id.X509CertificatePEM)
+	// Write files to local file system
+	writeFiles := func() error {
```

Comment:
I removed all the arguments for writeFiles() so that it stays consistent

Reply:
I believe it depends on whether if we're planning to call writeFiles() more than once or not.
If yes, pem data should be passed through arguments to avoid race in updating.

Reply:
@ctyano 
writeFiles() prolly won't run more than once at this point, and even the upcoming future, probably. 
It makes it simple like

prepareInstanceCert -> prepareRoleCert -> writeFiles() (if needed)

Reply:
@mlajkim 

>  writeFiles() prolly won't run more than once at this point, and even the upcoming future, probably.

I got the point and do agree that it's definitely better to have all the arguments removed.

Reply:
LGTM, mark as resolve